python/hail/methods/statgen.py

```diff
@@ -3028,12 +3034,12 @@ def ld_prune(call_expr, r2=0.2, bp_window_size=1000000, memory_per_core=256, kee
         0.0)
 
     std_gt_bm = BlockMatrix.from_entry_expr(standardized_mean_imputed_gt_expr, block_size)
+    r2_bm = (std_gt_bm @ std_gt_bm.T) ** 2
 
-    locally_pruned_positions = locally_pruned_table.key_by(contig=locally_pruned_table.locus.contig)
-    locally_pruned_positions = locally_pruned_positions.select(pos=locally_pruned_positions.locus.position)
+    _, stops = hl.locus_windows(locally_pruned_table.locus, bp_window_size)
 
-    entries = (std_gt_bm @ std_gt_bm.T)._filtered_entries_table(locally_pruned_positions, bp_window_size, False)
-    entries = entries.filter((entries.entry ** 2 >= r2) & (entries.i < entries.j))
+    entries = r2_bm.sparsify_row_intervals(range(0, stops.size), stops, blocks_only=True).entries()
```

Comment:
it's because in the entries table, we only want to include the edge between `i` and `j` once. We do this by enforcing that `i < j`, which corresponds to the upper triangle. In other words, `r2_bm` is a symmetric matrix and if `i < j` and the entry is at least the threshold, then we use `(i, j)` to represent the edge, not `(j, i)`.